### PR TITLE
Cover Unit test for OrderItemValidator

### PR DESCRIPTION
**Pull request by RahulJha-mastek**
Story [AB#13236](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13236)
Task [AB#14860](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/14860)

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs [hunk 1]
```diff
@@ -21,5 +28,122 @@ public static void Constructors_VerifyGuardClauses()
...
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+            order.CommencementDate.Should().NotBeNull();
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
+                new OrderItemRecipientModel { DeliveryDate = null }).ToList();
+
+            var result = orderItemValidator.Validate(order, new CreateOrderItemModel { ServiceRecipients = serviceRecipients }, CatalogueItemType.AdditionalService);
+
+            result.FailedValidations.Should().NotBeEmpty();
```
Comment (kiran-mastek) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs:82]: More explicit testing is desirable to check the right error is being set e.g.
```
foreach ((_, ValidationResult validationResult) in result.FailedValidations)
{
     var errorDetails = validationResult.Errors.SingleOrDefault();
     errorDetails.Should().NotBeNull();
     errorDetails.Field.Should().Be("DeliveryDate");
     errorDetails.Id.Should().Be("DeliveryDateRequired");
}
```

Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs:84]: Added more check in test cases.

**Review by kiran-mastek — DISMISSED** [review]

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs [hunk 2]
```diff
@@ -21,5 +29,151 @@ public static void Constructors_VerifyGuardClauses()
...
+        [InMemoryDbAutoData]
+        public static void Validate_NullDeliveryDate_SuccessIsFalse(
+            Order order,
+            CreateOrderItemModel model,
+            OrderItemValidator orderItemValidator)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+            order.CommencementDate.Should().NotBeNull();
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs:79]: As `r` isn't used we can use a discard here.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs [hunk 3]
```diff
@@ -21,5 +29,151 @@ public static void Constructors_VerifyGuardClauses()
...
+        [InMemoryDbAutoData]
+        public static void Validate_InvalidDeliveryDate_SuccessIsFalse(
+            Order order,
+            CreateOrderItemModel model,
+            CatalogueItemType itemType,
+            OrderItemValidator orderItemValidator)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs:107]: As above, as `r` isn't used we can use a discard here.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs [hunk 4]
```diff
@@ -21,5 +29,151 @@ public static void Constructors_VerifyGuardClauses()
...
+            Order order,
+            CreateOrderItemModel model,
+            CatalogueItemType itemType,
+            ValidationSettings validationSettings)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+
+            OrderItemValidator orderItemValidator = new OrderItemValidator(validationSettings);
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs:137]: As above, as `r` isn't used we can use a discard here.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs [hunk 5]
```diff
@@ -21,5 +29,151 @@ public static void Constructors_VerifyGuardClauses()
...
+            Order order,
+            CreateOrderItemModel model,
+            CatalogueItemType itemType,
+            ValidationSettings validationSettings)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+
+            itemType.Should().NotBeEquivalentTo(CatalogueItemType.AssociatedService);
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs:167]: As above, as `r` isn't used we can use a discard here.

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs (hunk 2 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs]: removed `r`

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs (hunk 3 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs]: removed `r`

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs (hunk 4 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs]: removed `r`

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs (hunk 5 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs]: removed `r`

**Pull request merged**